src/error/error-qvm.lisp

```diff
@@ -0,0 +1,142 @@
+;;;; error-qvm.lisp
+;;;;
+;;;; This file implements an efficient QVM which tracks the propagation of Pauli
+;;;; errors through a CNOT-dihedral circuit which, in the absence of errors, is
+;;;; always meant to MEASURE zeroes.  For example, surface code cycles for the
+;;;; usual toric code have this property.
+
+(in-package #:qvm.error)
+
+(defclass error-qvm (fowler-qvm classical-memory-mixin)
+  ((X-vector
+    :accessor error-qvm-X-vector
+    :type simple-bit-vector
+    :documentation "A bit-vector which tracks whether this qubit has experienced an X-error.")
+   (Z-vector
+    :accessor error-qvm-Z-vector
+    :type simple-bit-vector
+    :documentation "A bit-vector which tracks whether this qubit has experienced a Z-error.")
+   (num-qubits
+    :type unsigned-byte
+    :reader error-qvm-num-qubits
+    :initarg :num-qubits))
+  (:documentation "A QVM that can efficiently the propagation of Pauli errors through gates drawn from the CNOT-dihedral group of order 8."))
+
+(defun make-error-qvm (num-qubits
+                       &key
+                         (classical-memory-model quil::**empty-memory-model**)
+                         (noise-probability 0d0)
+                         (noise-class 0))
```

Comment:
where is this documented?

Reply:
Added here and to the docstring on the error-qvm class